astropy/utils/shapes.py

```diff
@@ -138,7 +149,13 @@ def squeeze(self, *args, **kwargs):
         """
         return self._apply("squeeze", *args, **kwargs)
 
-    def take(self, indices, axis=None, out=None, mode="raise"):
+    def take(
```

Comment:
In principle, these methods are meant to be similar to `ndarray`, but they look different from what numpy does: https://github.com/numpy/numpy/blob/2d2ff2501ad8a558c9ea402bf7b1a48e3f676fed/numpy/__init__.pyi#L1673-L1696

Note that I mention this mostly in case it matters; for me the numpy stuff is essentially ununderstandable, while this is quite clear, so from that point of view I prefer this!

Reply:
I did not look at any type stubs, the annotations that I added to these methods are my transcriptions of docstrings. But because you brought up the type stubs it might be good to comment on at least one of them.

For `take()` `numpy` defines three overloads. The third one is for the case where `out` is not `None`, but our method raises a `NotImplementedError` in that case, so that overload is not applicable for us. The remaining two distinguish calls that produce scalar values from calls that produce array values. For automatic type checking this distinction can be useful, but for helping IDEs perform tab-completion it is less important.

Reply:
Ah, thanks. For our situation, the scalar case is also irrelevant, since we still return the same class. 

I'm very happy to just aim for ease-of-reading and tab completion in this first iteration!